src/myrtlespeech/model/rnn.py

```diff
@@ -146,59 +144,58 @@ def forward(self, x: Tuple[RNNData, Lengths]) -> Tuple[RNNData, Lengths]:
                 sequence input and the second represents the length of these
                 *input* sequences.
 
+            hx: The Optional hidden RNNState.
+
         Returns:
-            A Tuple[RNNData, Lengths] where the first element is the rnn
-                sequence output and the second represents the length of these
-                *output* sequences. These lengths will be unchanged from the
-                input lengths.
-
-                The sequence output of :py:obj:`RNNData` will have the same
-                subtype as ``x[0]`` so, if the user would like the hidden state
-                returned at the start-of-sequence, they should pass a hidden
-                state of :py:data:`None` and PyTorch will initialise the
-                hidden state(s) to zero.
+            A Tuple[Tuple[RNNData, Lengths], RNNState] where the first element
+                is the rnn sequence output and the second represents the
+                length of these *output* sequences. These lengths will be
+                unchanged from the input lengths.
         """
-
-        if isinstance(x[0], torch.Tensor):
-            inp = x[0]
-            hid = None
-            return_tuple = False
-        elif isinstance(x[0], tuple) and len(x[0]) == 2:
-            inp, hid = x[0]
-            return_tuple = True
-        else:
-            raise ValueError("`x[0]` must be of type RNNData.")
+        inp, lengths = x
+
+        if hx is None:
+            # explicitly define hidden state for onnx export
+            num_directions = 2 if self.bidirectional else 1
+            zeros = torch.zeros(
+                self.rnn.num_layers * num_directions,
+                x[1].shape[0],
+                self.rnn.hidden_size,
+                dtype=inp.dtype,
+            )
+            if self.rnn_type == RNNType.LSTM:
+                hx = (zeros, zeros)
+            else:
+                hx = zeros
 
         if self.use_cuda:
             inp = inp.cuda()
-            if hid is not None:
-                if isinstance(hid, tuple) and len(hid) == 2:  # LSTM
-                    hid = hid[0].cuda(), hid[1].cuda()
-                elif isinstance(hid, torch.Tensor):  # Vanilla RNN/GRU
-                    hid = hid.cuda()
+            if hx is not None:
+                if isinstance(hx, tuple) and len(hx) == 2:  # LSTM
```

Comment:
Make the comment redundant? :-) 

```suggestion
                if self.rnn_type == RNNType.LSTM:
```

Reply:
Could change `elif` to `else` below also if this change is accepted.

Reply:
Nice! Done.